### PR TITLE
feat(esl-utils): add ability to reject promisifyEvent by using AbortSignal

### PR DESCRIPTION
**Pull request by dshovchko**
Closes: #2390

### CODE REVIEW

**Review by yadamskaya — APPROVED** [review]

File: src/modules/esl-utils/async/promise/event.ts [hunk 1]
```diff
@@ -33,6 +33,9 @@ export function promisifyEvent(
     if (typeof timeout === 'number' && timeout >= 0) {
       setTimeout(eventCallback, timeout);
     }
+    if (typeof options === 'object') {
+      options.signal?.addEventListener('abort', () => reject(new Error('Rejected by abort signal')), {once: true});
+    }
```
Comment (ala-n) [inline src/modules/esl-utils/async/promise/event.ts:38]: Memory leak: we have 2 cases when we end with signal subscription:
1) Negative (when original event observation is canceled by signal) - in that case we have an `abort` event and once will cover our requirement
2) Positive (the original Event was caught, so no signal involvement before event) - and here we end up with useless signal object that can not be garbage-collected as it have an active subscribtion.

File: src/modules/esl-utils/async/test/promise/event.test.ts [hunk 2]
```diff
@@ -20,11 +20,12 @@ describe('async/promise/event', () => {
...
     });
 
-    test('Rejected by timeout if it is exceeded', () => {
+    test('Rejected by timeout if it is exceeded', async () => {
       const el = document.createElement('div');
       const promise$ = promisifyEvent(el, 'test', 10);
       jest.advanceTimersByTime(100);
-      return expect(promise$).rejects.toBeInstanceOf(Error);
+      await expect(promise$).rejects.toBeInstanceOf(Error);
+      await expect(promise$).rejects.toThrow(new Error('Rejected by timeout'));
```
Comment (ala-n) [inline src/modules/esl-utils/async/test/promise/event.test.ts:28]:
```suggestion
      await expect(promise$).rejects.toThrow(new Error('Rejected by timeout'));
```

File: src/modules/esl-utils/async/test/promise/event.test.ts [hunk 3]
```diff
@@ -37,6 +38,58 @@ describe('async/promise/event', () => {
...
       }
     });
+    describe('AbortSignal handling', () => {
+      test('Rejected when was passed options with signal and invoked abort', async () => {
+        const el = document.createElement('div');
+        const controller = new AbortController();
+        const promise$ = promisifyEvent(el, 'test', null, {signal: controller.signal});
+        controller.abort();
+        await expect(promise$).rejects.toBeInstanceOf(Error);
+        await expect(promise$).rejects.toThrow(new Error('Rejected by abort signal'));
```
Comment (ala-n) [inline src/modules/esl-utils/async/test/promise/event.test.ts:48]:
```suggestion
        await expect(promise$).rejects.toThrow(new Error('Rejected by abort signal'));
```

File: src/modules/esl-utils/async/test/promise/event.test.ts [hunk 4]
```diff
@@ -37,6 +38,58 @@ describe('async/promise/event', () => {
...
+      });
+      test('Rejected when was passed options with signal in aborted state', async () => {
+        const el = document.createElement('div');
+        const promise$ = promisifyEvent(el, 'test', null, {signal: AbortSignal.abort()});
+        await expect(promise$).rejects.toBeInstanceOf(Error);
```
Comment (ala-n) [inline src/modules/esl-utils/async/test/promise/event.test.ts:53]:
```suggestion
```

**Review by ala-n — APPROVED** [review]

**Review by NastaLeo — APPROVED** [review]

**Review by fshovchko — APPROVED** [review]

**Pull request merged**